src/session/content/message_row/indicators.rs

```diff
@@ -93,12 +109,19 @@ mod imp {
         fn constructed(&self, obj: &Self::Type) {
             self.parent_constructed(obj);
 
-            self.0.connect_notify_local(
+            self.model.connect_notify_local(
                 Some("message"),
                 clone!(@weak obj => move |_, _| {
                     obj.notify("message");
                 }),
             );
+
+            Self::Type::this_expression("model")
+                .chain_property::<MessageIndicatorsModel>("sending-state-icon-name")
+                .chain_closure::<bool>(closure!(|_: glib::Object, icon_name: String| {
+                    !icon_name.is_empty()
+                }))
+                .bind(&*self.sending_state_icon, "visible", Some(obj));
```

Comment:
This can be simpler by just listening to the icon's `icon-name` notify signal and checking its value.